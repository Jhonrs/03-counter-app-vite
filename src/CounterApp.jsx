import { useState } from "react";
import PropsTypes from "prop-types";

export const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value);
    const handleAdd = () => {
        setCounter(counter + 1);
        //setCounter( (c) => c + 1 );
    };
    const handleRemove = () => setCounter(counter - 1);
    const handleReset = () => setCounter(value);

    return (
        <>
            <h1>CounterApp</h1>

            <h2>{counter}</h2>
            <button data-testid="btn-increase" onClick={handleAdd}>+1</button>
            <button data-testid="btn-decrease" onClick={handleRemove}>-1</button>
            <button data-testid="btn-reset" onClick={handleReset}>Reset</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropsTypes.number.isRequired,
};
