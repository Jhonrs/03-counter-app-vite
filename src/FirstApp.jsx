import PropsTypes from "prop-types";

export const FirstApp = ({ title, subtitle }) => {
  return (
    <>
      <h2>{title}</h2>
      <h6> {subtitle}</h6>
    </>
  );
}

FirstApp.propTypes = {
  title: PropsTypes.string.isRequired,
  subtitle: PropsTypes.string.isRequired,
}

FirstApp.defaultProps = {
  subtitle: 'Template del subtitulo',
  title: 'Template del titulo',
}