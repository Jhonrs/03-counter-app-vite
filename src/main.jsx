import React from 'react';
import ReactDOM from 'react-dom/client';
import {HelloWorldApp} from './HelloWorldApp';
import {CounterApp} from './CounterApp';
import './styles.css'
import { FirstApp } from './FirstApp';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
       {/*  <CounterApp value={6}/> */}
       <FirstApp title='Hola, soy Goku'/>
    </React.StrictMode>
)