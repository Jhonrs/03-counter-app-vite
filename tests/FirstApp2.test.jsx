import {
  getAllByAltText,
  getAllByText,
  getByText,
  render,
  screen
} from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <FirstApp />", () => {

    const title = 'Hola, soy Goku';
    const subTitle = 'Template del subtitulo';

    test('debe de hacer match con el snapshot', () => { 
        const { container } = render( <FirstApp title={title} />);
        expect ( container ).toMatchSnapshot();
    });

    test('debe de mostrar el mensaje "Hola, soy Goku"', () => { 
        render( <FirstApp title={title} />);
        expect( screen.getByText(title) ).toBeTruthy();
        /*  screen.debug(); */ /* sirve para ver el objeto */ 
    });
    
    test('debe buscarlo en un h1', () => { 
        render( <FirstApp title={title} />);
        expect( screen.getByRole('heading', { level: 2}).innerHTML ).toContain( title );
    });

    test('debe de mostrar el subtitulo enviado por props', () => { 
        render( 
            <FirstApp 
                title={ title }
                subTitle={ subTitle }
            />
        );
        expect( screen.getAllByText( subTitle ).length ).toBe(2);
    });
    
    
});
