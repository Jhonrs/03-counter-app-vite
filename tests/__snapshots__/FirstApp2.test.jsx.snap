// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h2
    data-testid="test-title"
  >
    Hola, soy Goku
  </h2>
  <h3>
     
    Template del subtitulo
  </h3>
  <h3>
     
    Template del subtitulo
  </h3>
  <h4>
     
    Template del nombre
  </h4>
</div>
`;
